app/models/piece.rb

```diff
@@ -104,4 +104,13 @@ def is_move_malformed?(new_rank, new_file)
     return true if new_file < 1 || new_file > 8
     false
   end
+
+  def horizontal_move?(new_rank, new_file)
+    rank == new_rank
+  end
+
+  def vertical_move?(new_rank, new_file)
+    file == new_file
+  end
```

Comment:
I think it's overkill to write a separate method for a single boolean comparison. The code would be easier to follow (no need to look at two separate files), more concise, and marginally more efficient if you just did this inline as part of `Rook#valid_move?`.